src/Microsoft.AspNetCore.Sockets/Internal/ChannelConnection.cs

```diff
@@ -6,24 +6,32 @@
 
 namespace Microsoft.AspNetCore.Sockets.Internal
 {
+    public static class ChannelConnection
```

Comment:
Because it doesn't infer. The whole point of this is you can infer the `T`. But inference only works with method type parameters. Hence [`Task.FromResult`](https://referencesource.microsoft.com/#mscorlib/system/threading/Tasks/Task.cs,11a386e7d7cae64a), [`Tuple.Create`](https://referencesource.microsoft.com/#mscorlib/system/tuple.cs,9124c4bea9ab0199), etc.

If we didn't care about inference, we'd just keep using the constructor.